### PR TITLE
Adjusted hook tests to work on windows

### LINKED ISSUE #235: Windows commit hooks without wsl
Current ci seems broken due to lack of wsl support and it turns out that without wsl our commit hook unit tests won't work.

It would be great if someone with access to a windows machine looks at this and makes them run able on vanilla windows 🤞

### PR DESCRIPTION
**Pull request by dr-BEat**
This PR fixes #235.
The extra newline at the beginning of the hook caused the following error on windows:
`error: cannot spawn .git/hooks/commit-msg: No such file or directory`

### CODE REVIEW

**Comment by dr-BEat** [comment]
Apparently my local setup is different from the CI system.
It still returns the same error: `Windows Subsystem for Linux has no installed distributions.\r\r\nDistributions can be installed by visiting the Microsoft Store:\r\r\nhttps://aka.ms/wslstore\r\r\n`

**Comment by extrawurst** [comment]
@dr-BEat thanks for looking into it. Is the wsl default installed on every Windows nowadays? Or do we have to simulate it by using a vanilla windows without wsl?

**Comment by dr-BEat** [comment]
@extrawurst no problem. Its not installed by default, however I think they added a `bash.exe` by default that gives the error above.
Usually any git installation on Windows brings a real bash, so it works out normally.
The `run_hook` functions in hooks.rs tries to call `bash` directly and it gets the fake Windows one.

**Comment by extrawurst** [comment]
Any idea how to fix this? Maybe just install some git using chocolaty in the ci?

**Comment by dr-BEat** [comment]
Yeah installing git might fix it.

**Comment by dr-BEat** [comment]
Gits already installed in the CI image. (:
https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md

**Comment by dr-BEat** [comment]
According to this blog by Jon Skeet https://codeblog.jonskeet.uk/2019/10/12/using-git-bash-from-appveyor/ the order of directories in the Path environment can break stuff. 
I tried to add the git path as the first entry, but it does not seem to work.

**Comment by extrawurst** [comment]
@dr-BEat any more ideas?

**Comment by dr-BEat** [comment]
@extrawurst nothing concrete but I will try some tricks, probably the path is not correct still

**Comment by dr-BEat** [comment]
@extrawurst the path environment is correct, the problem is that `Command::new("bash")` in hooks.rs line 105 somehow does not follow normal windows rules in finding which bash to execute.
If I hardcode the full path to the git bash with `Command::new(r"C:\Program Files\Git\bin\bash.exe")` it works perfectly.

**Comment by extrawurst** [comment]
@dr-BEat interesting. maybe because we run the `cargo test` in the git shell, did you try setting the `shell: cmd` in the ci to make him use the windows stuff?

**Comment by dr-BEat** [comment]
@extrawurst I do not think its related to the shell. I can reproduce the issue locally when just running cargo test from Powershell

**Comment by extrawurst** [comment]
@dr-BEat nice a repro is what we needed!!🥳 I wish I could add any windows knowledge whatsoever :(

**Comment by dr-BEat** [comment]
@extrawurst found it! Calling `env` on the command fixed it. This is probably related to the discussion here https://github.com/rust-lang/rust/issues/37519. The behaviour of `Command::spawn()` changes because the environment is passed differently to Windows if `env` is called.

**Comment by extrawurst** [comment]
OMFG!!! you are the man!!

File: asyncgit/src/sync/hooks.rs [hunk 1]
```diff
@@ -105,6 +105,7 @@ fn run_hook(
     let output = Command::new("bash")
         .args(bash_args)
         .current_dir(path)
+        .env("DUMMYENV", "FixPathHandlingOnWindows") // This call forces Command to handle the Path environment correctly on windows, the specific env set here does not matter
```
Comment (extrawurst) [inline asyncgit/src/sync/hooks.rs:108]: wtf... this is crazy! 🙈

**Pull request merged**